app/cmd/profiler.go

```diff
@@ -0,0 +1,250 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+
+	replicaClient "github.com/longhorn/longhorn-engine/pkg/replica/client"
+)
+
+const (
+	opShow    = "show"
+	opEnable  = "enable"
+	opDisable = "disable"
+)
+
+func ProfilerCmd() cli.Command {
+	return cli.Command{
+		Name: "profiler",
+		Flags: []cli.Flag{
+			cli.BoolFlag{
+				Name:  "engine",
+				Usage: "Execute profiler command on engine",
+			},
+			cli.BoolFlag{
+				Name:  "replica",
+				Usage: "Execute profiler command on replica",
+			},
+		},
+		Subcommands: []cli.Command{
+			ProfilerShowCmd(),
+			ProfilerEnableCmd(),
+			ProfilerDisableCmd(),
+		},
+	}
+}
+
+func ProfilerShowCmd() cli.Command {
+	return cli.Command{
+		Name: "show",
+		Action: func(c *cli.Context) {
+			if err := showProfiler(c); err != nil {
+				logrus.WithError(err).Fatalf("Error running profiler show")
+			}
+		},
+	}
+}
+
+func ProfilerEnableCmd() cli.Command {
+	return cli.Command{
+		Name: "enable",
+		Flags: []cli.Flag{
+			cli.IntFlag{
+				Name:     "port",
+				Hidden:   false,
+				Required: false,
+				Value:    0,
+				Usage:    "run profiler with specific port. The port number should bigger than 30000.",
+			},
+		},
+		Action: func(c *cli.Context) {
+			if err := enableProfiler(c); err != nil {
+				logrus.WithError(err).Fatalf("Error running profiler enable")
+			}
+		},
+	}
+}
+
+func ProfilerDisableCmd() cli.Command {
+	return cli.Command{
+		Name: "disable",
+		Action: func(c *cli.Context) {
+			if err := disableProfiler(c); err != nil {
+				logrus.WithError(err).Fatalf("Error running profiler disable")
+			}
+		},
+	}
+}
+
+func opValidate(engine, replica bool) bool {
```

Comment:
`opValidate ....) error {`

This is a check function, so suggest returning an error instead of a bool var to have more info about why the check would fail.